ui/lib/kv/addon/components/kv-delete-modal.js

```diff
@@ -0,0 +1,134 @@
+      case 'delete metadata':
+        return 'This will permanently delete the metadata and versions of the secret. All version history will be removed. This cannot be undone.';
+      default:
+        return assert('mode must be one of undelete, delete, destroy, delete metadata.');
+    }
+  }
+
+  get generateDeleteRadioOptions() {
+    let isDeactivated = false;
+    if (this.args.secret.canReadMetadata) {
+      isDeactivated = this.args.metadata?.currentSecret.isDeactivated;
+    }
+
+    return [
+      {
+        key: 'delete-version',
+        label: 'Delete this version',
+        description: `This deletes Version ${this.args.secret.version} of the secret.`,
+        disabled: !this.args.secret.canDeleteVersion,
+        tooltipMessage: 'You do not have permission to delete a specific version.',
+      },
+      {
+        key: 'delete-latest-version',
+        label: 'Delete latest version',
+        description: 'This deletes the most recent version of the secret.',
+        disabled: !this.args.secret.canDeleteLatestVersion || isDeactivated,
+        tooltipMessage: isDeactivated
+          ? `The latest version of the secret is already ${this.args.metadata.secretState.state}.`
+          : 'You do not have permission to delete the latest version of this secret.',
+      },
+    ];
+  }
+
+  @action handleButtonClick(mode) {
+    // undelete is the only use case were a modal is not shown.
+    if (mode === 'undelete') {
+      this.deleteType = 'undelete-version';
+      this.save.perform();
+      return;
+    }
+    this.modalOpen = true;
+    // deleteType is the param used in the case/switch for deleteRecord on the adapter.
+    if (mode === 'destroy') {
+      this.deleteType = 'destroy-version';
+    }
+    if (mode === 'delete metadata') {
+      this.deleteType = 'delete-metadata';
+    }
+    // if mode is delete, they still need to select what kind of delete operation they'd like to perform so we don't set the deleteType until they select a radio option.
+  }
+
+  flashMessageMessage(isSuccess) {
+    if (isSuccess) {
+      if (this.deleteType === 'delete-metadata') {
+        return `Successfully deleted the metadata and all version data for the secret ${this.args.path}.`;
+      } else {
+        return `Successfully ${verbToPastTense(this.args.mode, 'past')} Version ${
+          this.args.secret.version
+        } of ${this.args.path}.`;
+      }
+    } else {
+      if (this.deleteType === 'delete-metadata') {
+        return `There was an issue deleting ${this.args.path} metadata.`;
+      } else {
+        return `There was an issue ${verbToPastTense(this.args.mode, 'gerund')} Version ${
+          this.args.secret.version
+        } of ${this.args.path}.`;
+      }
+    }
+  }
+
+  @(task(function* () {
+    this.modalOpen = false;
+    try {
+      yield this.args.secret.destroyRecord({
+        adapterOptions: { deleteType: this.deleteType, deleteVersions: this.args.secret.version },
```

Comment:
For `delete-metadata` and `destroy-version` can we have `@mode` to be the same key we need in the adapter? That way we can remove the logic above setting `deleteType` - which can just be responsible for handling the two deleteTypes (when `@mode="delete"`)


it's so easy to get mixed up with which action maps to what endpoint so however we can streamline that would help our future selves!

```suggestion
      const deleteType = this.args.mode === 'delete' ? this.deleteType : this.args.mode 
      yield this.args.secret.destroyRecord({
        adapterOptions: { deleteType: deleteType, deleteVersions: this.args.secret.version },
```